lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -2,42 +2,73 @@ module ActiveModel
   class Serializer
     module Adapter
       class Attributes < Base
+        attr_accessor :_nested_associations
+
+        def initialize(serializer, options = {})
+          options ||= {}
+
+          self._nested_associations = ActiveModel::Serializer::Utils
+            .include_args_to_hash(options[:include])
+
+          super(serializer, options)
```

Comment:
@beauby there's basically two public methods on an adapter `::new` and `#serializable_hash`.  Yeah, some other stuff is public, it's really for AMS.  And AMS controls everything that happens between those two methods.